bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/api/EnedisHttpApi.java

```diff
@@ -264,22 +241,59 @@ private Consumption getMeasures(String userId, String prmId, LocalDate from, Loc
         }
         logger.trace("getData returned {}", data);
         try {
-            ConsumptionReport report = gson.fromJson(data, ConsumptionReport.class);
-            if (report == null) {
+            MeterResponse meterResponse = gson.fromJson(data, MeterResponse.class);
+            if (meterResponse == null) {
                 throw new LinkyException("No report data received");
             }
-            return report.firstLevel.consumptions;
+            return meterResponse.meterReading;
         } catch (JsonSyntaxException e) {
             logger.debug("invalid JSON response not matching ConsumptionReport.class: {}", data);
             throw new LinkyException(e, "Requesting '%s' returned an invalid JSON response", url);
         }
     }
 
-    public Consumption getEnergyData(String userId, String prmId, LocalDate from, LocalDate to) throws LinkyException {
-        return getMeasures(userId, prmId, from, to, "energie");
+    public MeterReading getEnergyData(String userId, String prmId, LocalDate from, LocalDate to) throws LinkyException {
+        return getMeasures(userId, prmId, from, to, "daily_consumption");
+    }
+
+    public MeterReading getPowerData(String userId, String prmId, LocalDate from, LocalDate to) throws LinkyException {
+        return getMeasures(userId, prmId, from, to, "daily_consumption_max_power");
     }
 
-    public Consumption getPowerData(String userId, String prmId, LocalDate from, LocalDate to) throws LinkyException {
-        return getMeasures(userId, prmId, from, to, "pmax");
+    public String getTempoData() throws LinkyException {
+        String url = String.format(TEMPO_URL, "2024-01-01", "2024-01-31");
+        if (!connected) {
+            initialize();
+        }
+        String data = getData(url);
+        if (data.isEmpty()) {
+            throw new LinkyException("Requesting '%s' returned an empty response", url);
+        }
+        logger.trace("getData returned {}", data);
+
+        try {
+            TempoResponse tempResponse = gson.fromJson(data, TempoResponse.class);
+            if (tempResponse == null) {
+                throw new LinkyException("No report data received");
+            }
+
+            return "{\"2024-01-20\":\"WHITE\",\"2024-01-21\":\"RED\",\"array\":[\"2024-01-20\",\"2024-01-21\"]}";
+            // return tempResponse.tempoDayInfo;
```

Comment:
Tempo data could probably take advantage of the new TimeSeries introduced

Reply:
Yes surrelly.
But if ok for you, I think we can introduce TimeSeries in next iterations

Reply:
tempo handling was completly review in last version, using timeseries, and better handling type on the channel